radix-engine-stores/src/hash_tree/test.rs

```diff
@@ -124,37 +127,82 @@ fn supports_empty_state() {
     assert_eq!(hash_v3, SPARSE_MERKLE_PLACEHOLDER_HASH);
 }
 
-// Note: this test relies on the impl details of underlying tree structure.
-// In particular, it is now assuming a single tree, keyed by hash(substate_id).
-// The asserts will need to be adjusted after introducing separate ReNode tree
-// vs Substate trees.
 #[test]
-#[ignore]
```

Comment:
Note from author: this test was very detailed and cool, but unfortunately also fragile (sensitive to how we do hashing of substate ids). Thank you @talekhinezh for tracking it down and disabling 👍 
I replaced the assert with a more coarse grained one (i.e. "some previous version nodes are marked as stale") and re-enabled.